### ISSUE TITLE
open youtube links with newpipe

### ISSUE THREAD

**Issue by cup** [TeamNewPipe/NewPipe#2160]
is this something this app is supposed to do?

because i was assuming if i clicked a youtube link in either chrome or firefox that it would automatically open in newpipe

not only does it not do that - but i dont see an option for it in the settings - and long press doesnt offer the option either

**Comment by TobiGr** [comment]
You can share the link with NewPipe

**Comment by AnotherLife** [comment]
I get a menu of apps and select Newpipe

**Comment by snappyapple632** [comment]
For now, you can open links with NewPipe using the share function. I'm not sure if we'll be getting an option for NewPipe as a "Default App" as you are describing, though.

**Comment by gkeegan** [comment]
I actually have NewPipe open YouTube links directly. I simply tap the URL, then it will give the menu of whether to see the video page, play the video, play in background, etc. I am unsure of what I did to get it to that point, but I would assume it is because I do not have the YouTube app installed (and thus no other default handler for YouTube links).

**Comment by Phoenix616** [comment]
It works like this for me and I have the official YouTube app installed. Make sure that it isn't set as the default. That can be done by going to your YouTub app's application info settings page and clearing the "open by default" setting. Then a menu should pop up where you can select the app when you click a link to youtube.

**Comment by cup** [comment]
I did get this to work with Chrome

with Firefox it seems to only work on long press:

https://bugzilla.mozilla.org/show_bug.cgi?id=806385

I think what fixed it was

Settings, Apps & Notifications, YouTube, Open by default, CLEAR DEFAULTS

**Comment by alshanu** [comment]
I did this 
Settings, Apps & Notifications, YouTube, Open supported links, don't open in this app. After that when I tried to open youtube like from chrome it showed option to open in newpipe.

**Comment by Nuc1eoN** [comment]
Hi, I do have have the official youtube app installed and I am not getting an app selection menu when tapping on chrome youtube links.

I cannot clear anything since it doesn't exist :) 
How would I configure NewPipe in that case?

**Comment by opusforlife2** [comment]
@Nuc1eoN Does Chrome have a setting to open links in apps? Firefox does. See if there is a setting like that you can toggle.

**Comment by benmordecai** [comment]
Note that in Grapheneos and probably others there is a system setting for default apps > opening links > newpipe > add link

**Comment by opusforlife2** [comment]
Yep. That's an Android 12 feature, not specific to custom ROMs.

**Comment by Nuc1eoN** [comment]
> @Nuc1eoN Does Chrome have a setting to open links in apps? Firefox does. See if there is a setting like that you can toggle.

I do not think it has that, sadly. Could not find such a setting.


> Note that in Grapheneos and probably others there is a system setting for default apps > opening links > newpipe > add link

Yes true, good tip!
Sadly it does has has no effect on links opened in the Browser, though.

**Comment by opusforlife2** [comment]
Then you could try using Firefox.

**Comment by Nuc1eoN** [comment]
> Then you could try using Firefox.

I think I will stay with Brave browser.

**Comment by opusforlife2** [comment]
Could you paste an example link here?

**Comment by benmordecai** [comment]
Test YouTube link: https://youtu.be/7ayg4shWKkw

**Comment by benmordecai** [comment]
Note that clicking the above link from Brave with the setting I mentioned in Graphene takes me to NewPipe. However, navigating to m.YouTube.com will allow you to play from Brave Broeser

**Comment by opusforlife2** [comment]
> I uninstalled all other YouTube apps, and now it just opens it straight in firefox.

@TheAllSeeing Firefox also has an "Open in apps" toggle. You could try that. If even that fails, although it might be painful in the short term, you could try clearing Firefox's app opening defaults as well and see if Newpipe opens directly.

@Nuc1eoN Similarly, you could try clearing defaults for Chrome.

**Comment by paulverbeke** [comment]
@opusforlife2  so with Android 12 isn't there a way for NewPipe to auto check all available links ? Or ask the user to do it and then redirecting them to the correct page ? Because it happened to me on GrapheneOS. YouTube not installed, and Firefox with "open links in apps" enabled. Never received any popup and YouTube links always opening in Firefox. I had to manually go check all the 50 links.

**Comment by opusforlife2** [comment]
That's just how it is with Android 12, sadly.

**Comment by paulverbeke** [comment]
Yes I since dived into the technicalities of this new feature and I see why third party apps now can't automatically use domains they do not own. It's quite elegant on Google's part I must say.
The sad thing is that by upgrading to 12 it broke some of my workflows without explicit warning post upgrade.
I just noticed it's the same for FastHub, ViMusic, etc...

Couldn't third party app like Newpipe have a popup to explain, instruct, and eventually redirect to the "open links" settings ?
